components/pagerduty/actions/find-oncall-user/find-oncall-user.mjs

```diff
@@ -0,0 +1,47 @@
+import pagerduty from "../../pagerduty.app.mjs";
+
+export default {
+  key: "pagerduty-find-oncall-user",
+  name: "Find Oncall User",
+  description: "Find the user on call for a specific schedule. [See the docs here](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODE0MA-create-an-incident)",
+  type: "action",
+  version: "0.0.1",
+  props: {
+    pagerduty,
+    scheduleId: {
+      propDefinition: [
+        pagerduty,
+        "oncallScheduleId",
+      ],
+    },
+    userId: {
+      optional: false,
+      propDefinition: [
+        pagerduty,
+        "userId",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const {
+      scheduleId,
+      userId,
+    } = this;
+
+    const { users } =
+      await this.pagerduty.listUsersOncall({
+        $,
+        scheduleId,
+      });
+
+    const userFound = users.find(({ id }) => id === userId);
+
+    const summary = userFound
+      ? `${userFound.name} is On-Call for selected schedule`
+      : "No user found for selected schedule";
```

Comment:
**Question**
I'm not sure about this, but from what I understood should this action return the user on call (or users) for a schedule instead of checking if a user is on call?

Reply:
Ok since the description says **Find user on call**  I assumed it should be for a single user and not for a list of users, so that's why the *User ID* is required as well

Reply:
I found this action name to be misleading compared to what it's doing.

It should be called "find IF user is on-call" instead of "find on-call user".